metadata-ingestion/src/datahub/ingestion/source/unity/source.py

```diff
@@ -187,10 +193,21 @@ def get_workunits_internal(self) -> Iterable[MetadataWorkUnit]:
                 )
                 return
 
+        self.report.report_ingestion_stage_start("Ingest service principals")
         self.build_service_principal_map()
+        if self.config.include_notebooks:
+            self.report.report_ingestion_stage_start("Ingest notebooks")
+            self.build_notebook_map()
+
         yield from self.process_metastores()
 
+        if self.config.include_notebooks:
```

Comment:
Can't we move line 198 to here and then we don't need to separate the notebook processing logic.

Reply:
No, I need to set the notebook's upstreams before I can generate the workunits for it. I suppose I can generate the lineage mcp afterwards... I'll do that